lib/parser/onnx/onnx_parser.cc

```diff
@@ -404,42 +412,56 @@ Status ONNXParser::ConvertConstNode(const onnx::NodeProto& cur_node) {
   return Status::SUCCESS;
 }
 
-Status ONNXParser::ConvertPlaceholderNode(
-    const onnx::ValueInfoProto& value_info_def) {
+Type ONNXParser::GetType(const onnx::ValueInfoProto& value_info_def) {
   HLCHECK(value_info_def.type().has_tensor_type() &&
           "Unsupported value info type.");
   auto type_def = value_info_def.type().tensor_type();
   DataType data_type = ProcessDataType(type_def.elem_type());
   auto shape_def = type_def.shape();
   const int dim_size = shape_def.dim_size();
   std::vector<int64_t> shape;
-  for (int i = 0; i < dim_size; i++) {
+  for (int i = 0; i < dim_size; ++i) {
     auto dim_def = shape_def.dim(i);
     if (dim_def.dim_value()) {
       shape.push_back(dim_def.dim_value());
     } else {
-      // TODO(unknown): Handle dim_param case
       shape.push_back(-1);
     }
   }
-  auto arg = arg_builder_->CreateArgument(value_info_def.name(),
-                                          Type(data_type, shape));
+  return Type(data_type, shape);
+}
+
+Status ONNXParser::ConvertSubPlaceholderNode(
+    ArgumentBuilder* arg_builder, const onnx::ValueInfoProto& value_info_def) {
+  HLCHECK(!loop_arg_types_.empty());
+  auto& type = loop_arg_types_.top();
+  if (!type.IsValid()) {
+    type = GetType(value_info_def);
```

Comment:
why not just use the type in value_info_def?

Reply:
value_info_def in loop body maybe invalid:
1.can get data type, shape not define
2.can get shape, data type not define
3.both is not define
 but we can use stack to get the type from loop node